### ISSUE TITLE
UV not rendering after split

### ISSUE THREAD

**Issue by KatharineV** [scientist-softserv/adventist-dl#748]
Works uploaded after Knapsack are not rendering in the UV after the pages are split. We expect all split documents to render in the UV. 

In this screenshot from SDAPI prod, you see the first split page shows in the Items list, but the document doesn't render in PDF.js (see #746) or the UV. 

https://sdapi.b2.adventistdigitallibrary.org/concern/journal_articles/dialogue_2023_2_29_31_a_pioneer_of_adventist_church_school_curriculum_alma
![Image](https://github.com/scientist-softserv/adventist-dl/assets/118196520/c28da5d3-6f83-4352-8087-c6424e276d03)

**Comment by ShanaLMoore** [comment]
Local: This may be the issue - I created a Generic Work and attached a multi page pdf. The pdfs are attached but the UV displays an error. Yarn did not fix it. 



![Image](https://github.com/scientist-softserv/adventist-dl/assets/10081604/8fa103f2-8f57-4ae4-92c7-cd0a404b5857)

**Comment by ShanaLMoore** [comment]
PR started: 

https://github.com/samvera/hyku/pull/2185 

# TODO: 
Update Adventist Knapsack submodule once it gets merged into HYKU/adventist_dev

**Comment by ShanaLMoore** [comment]
You can see the UV trying to load after the changes in the above PR. I will turn PDF.js off and reprocess the works and report back if they render in the UV. 



![Image](https://github.com/scientist-softserv/adventist-dl/assets/10081604/a86e36e0-5ee8-4f31-aff1-737bc47fb095)

**Comment by ShanaLMoore** [comment]
QA: Pass ✅ 



![Image](https://github.com/scientist-softserv/adventist-dl/assets/10081604/17ac1407-7068-4df1-bd17-094b8f4af4d2)

**Comment by KatharineV** [comment]
Team, I'm having a heck of a time trying to test this ticket. I can't get a new work into either ADL or SDAPI production tenants, because of bugs that are blocking single upload and bulkrax imports. So I can't see if a new work will split and render in the UV as expected. I can confirm that uploads from about a week ago or earlier this week are either a) [not showing](https://sdapi.b2.adventistdigitallibrary.org/concern/journal_articles/dialogue_2023_2_25_28_carlot_dorve_a_dialogue_with_an_adventist_trumpeter) in the UV or _any _viewer,__ despite splitting, or b) [not splitting](https://adl.b2.adventistdigitallibrary.org/concern/published_works/22266514_cardinal_2009) at all and just stopped at the PDF.js stage. 

On a staging site, I managed to use Bulkrax to bring in a work via OAI, and it has not split yet (see link below). It's been a couple hours since import. No splitting, no UV...so I still can't test. I feel pretty stuck. It seems like this ticket isn't fixed, based on older works, but I can't confirm with a new upload until other blockers are cleared...unless you have a suggestion for me.

https://testing.s2.adventistdigitallibrary.org/concern/published_works/20121633_to_brother_j_n_andrews_and_sister_h_n_smith?locale=en

**Comment by KatharineV** [comment]
Related to this ticket, I found a work that has child works visible via the Dashboard, but they don't show on the parent record in the items list. The work is not loading in the UV, the child works seem lost, and even PDF.js isn't rendering right now (it's been spinning for a few minutes).

https://adl.b2.adventistdigitallibrary.org/concern/published_works/22266530_cardinal_2017

No child works on the work page:


![Image](https://github.com/scientist-softserv/adventist-dl/assets/118196520/b08808fb-aeef-486b-b6a1-ae761dfcfb0a)

Child works do show on the Dashboard:


![Image](https://github.com/scientist-softserv/adventist-dl/assets/118196520/e199deba-1503-4c83-8b76-c142fb51704d)

Here's one of them: https://adl.b2.adventistdigitallibrary.org/concern/published_works/6c63102e-2e93-40bc-84df-e7fcd46ee127?locale=en

**Comment by ShanaLMoore** [comment]
Hi @KatharineV Is this still an issue? ref: https://adl.b2.adventistdigitallibrary.org/concern/published_works/22266530_cardinal_2017

**Comment by KatharineV** [comment]
Confirmed that the work that was stuck before is rendering in the UV now. https://adl.b2.adventistdigitallibrary.org/concern/published_works/22266530_cardinal_2017 

Hooray! 

![Image](https://github.com/scientist-softserv/adventist-dl/assets/118196520/18110cac-ef17-4e92-8795-760888829801)

I also just uploaded a set of works to SDAPI prod using a CSV/files importer and the PDFs split and are displaying as expected in the UV:

https://sdapi.b2.adventistdigitallibrary.org/concern/journal_articles/dynamic_steward_2023_04_06_18_19_proposal_of_partnership_between_treasury_a

**Comment by KatharineV** [comment]
Team, I found a work today where the UV isn't rendering anything on ADL prod: https://adl.b2.adventistdigitallibrary.org/concern/images/20000208_illustration_of_harmon_hall. The attached files are images, so I expect to see the UV, not PDF.js. I see nothing. Is this a one-off issue or possibly related to the UV bug, post-Knapsack? Thanks for checking!
@ShanaLMoore I hope you don't mind me tagging you. This ticket moved to the Done category, so I want to make sure this note is seen. 


Here's how the work is displaying for me right now:

![Image](https://github.com/scientist-softserv/adventist-dl/assets/118196520/0dc6d802-9a68-4ab6-b44c-cffbc12dfc2e)

**Comment by ShanaLMoore** [comment]
@KatharineV Do you know how this record was created?

**Comment by KatharineV** [comment]
Yes, this is a work that my predecessor uploaded using Bulkrax and the OAI feed (adl:image set, I believe) back in December 2021. It would be part of the earliest batch uploads. Other works from those loads do show in the UV, which is why I'm suspicious this might be an issue with the work, not Knapsack. 

I see your edits now. Let me try updating that single record and TIFF!
